selfservice/strategy/password/login_test.go

```diff
@@ -864,4 +860,168 @@ func TestCompleteLogin(t *testing.T) {
 			false, true, http.StatusOK, redirTS.URL)
 		assert.Equal(t, identifier, gjson.Get(body, "identity.traits.subject").String(), "%s", body)
 	})
+
+	t.Run("suite=password migration hook", func(t *testing.T) {
+		ctx := context.Background()
+
+		passwordDB := make(map[string]string)
+
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			b, err := io.ReadAll(r.Body)
+			require.NoError(t, err)
+			_ = r.Body.Close()
+			var payload struct {
+				Identifier string `json:"identifier"`
+				Password   string `json:"password"`
+			}
+			require.NoError(t, json.Unmarshal(b, &payload))
+
+			switch pw := passwordDB[payload.Identifier]; pw {
```

Comment:
Looks like we're missing a case for the server responding with a 200 status code, but unexpected body.